File: app/src/main/java/com/tunasteak/activity/ExpandableListViewBindingActivity.java
package com.tunasteak.activity;

import android.view.View;

import com.tunasteak.R;
import com.tunasteak.databinding.ActivityExpandableListViewBinding;
import com.tunasteak.model.ExpandableListViewModel;
import com.tunasteak.model.TBindingModel;

public class ExpandableListViewBindingActivity extends TBindingActivity<ActivityExpandableListViewBinding> {
    @Override
    protected int getLayoutId() {
        return R.layout.activity_expandable_list_view;
    }

    @Override
    protected TBindingModel getModel() {
        return new ExpandableListViewModel(binding, getIntent());
    }

    @Override
    public void onInit() {
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            default:
                break;
        }
    }
}

File: app/src/main/java/com/tunasteak/activity/ExpandableListViewBindingActivity.kt
package com.tunasteak.activity

import android.view.View
import com.tunasteak.data.Data
import com.tunasteak.demo.R
import com.tunasteak.demo.databinding.ActivityExpandableListViewBinding
import com.tunasteak.model.TBindingModel
import com.tunasteak.model.ExpandableListViewModel

class ExpandableListViewBindingActivity : TBindingActivity<ActivityExpandableListViewBinding?>() {
    override fun getLayoutId(): Int {
        return R.layout.activity_expandable_list_view
    }

    override fun getModel(): TBindingModel<*> {
        return ExpandableListViewModel(binding, intent)
    }

    override fun getData(): Any {
        return Data.getInstance()
    }

    override fun onInit() {}
    override fun onClick(v: View) {
        when (v.id) {
            else -> {
            }
        }
    }
}